pkg/pxc/app/statefulset/haproxy.go

```diff
@@ -0,0 +1,218 @@
+	initDelay := int32(10)
+	if spec.LivenessInitialDelaySeconds != nil {
+		initDelay = *spec.LivenessInitialDelaySeconds
+	}
+	appc := corev1.Container{
+		Name:            haproxyName,
+		Image:           spec.Image,
+		ImagePullPolicy: corev1.PullAlways,
+		Ports: []corev1.ContainerPort{
+			{
+				ContainerPort: 3306,
+				Name:          "mysql",
+			},
+			{
+				ContainerPort: 1024,
+				Name:          "stat",
+			},
+		},
+		TerminationMessagePath: "/dev/termination-log",
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "haproxy-cfg",
+				MountPath: "/etc/haproxy/pxc",
+			},
+			{
+				Name:      "haproxy-auto",
+				MountPath: "/etc/haproxy-auto/",
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "POD_NAME",
+				Value: c.sfs.ObjectMeta.Name,
+			},
+			{
+				Name:  "POD_NAMESPACE",
+				Value: c.sfs.ObjectMeta.Namespace,
+			},
+			{
+				Name:  "PXC_SERVICE",
+				Value: c.labels["app.kubernetes.io/instance"] + "-" + "pxc",
+			},
+			{
+				Name: "MONITOR_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: app.SecretKeySelector("my-cluster-secrets", "monitor"),
+				},
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       10,
+			InitialDelaySeconds: initDelay,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/healthz",
+					Port:   intstr.FromInt(1024),
+					Scheme: "HTTP",
+				},
+			},
+		},
+		SecurityContext: spec.ContainerSecurityContext,
+	}
+
+	res, err := app.CreateResources(spec.Resources)
+	if err != nil {
+		return appc, fmt.Errorf("create resources error: %v", err)
+	}
+	appc.Resources = res
+
+	return appc, nil
+}
+
+func (c *HAProxy) SidecarContainers(spec *api.PodSpec, secrets string) ([]corev1.Container, error) {
+	res, err := app.CreateResources(spec.SidecarResources)
+	if err != nil {
+		return nil, fmt.Errorf("create sidecar resources error: %v", err)
+	}
+	return []corev1.Container{
+		{
+			Name:            "pxc-monit",
+			Image:           spec.Image,
+			ImagePullPolicy: corev1.PullAlways,
+			Args: []string{
+				"/usr/bin/peer-list",
+				"-on-change=/usr/bin/add_pxc_nodes.sh",
+				"-service=$(PXC_SERVICE)",
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "PXC_SERVICE",
+					Value: c.labels["app.kubernetes.io/instance"] + "-" + "pxc",
+				},
+				{
+					Name: "MONITOR_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: app.SecretKeySelector("my-cluster-secrets", "monitor"),
+					},
+				},
+			},
+			Resources:              res,
+			TerminationMessagePath: "/dev/termination-log",
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "haproxy-cfg",
+					MountPath: "/etc/haproxy/pxc",
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Add: []corev1.Capability{"SYS_PTRACE"},
+				},
+			},
```

Comment:
it should be passed from CR
```suggestion
			SecurityContext: spec.ContainerSecurityContext,
```